src/Core/src/Handlers/HybridWebView/HybridWebViewHandler.Android.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Android.Content;
+using Android.Webkit;
+using Java.Interop;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using static Android.Views.ViewGroup;
+using AWebView = Android.Webkit.WebView;
+using AUri = Android.Net.Uri;
+
+namespace Microsoft.Maui.Handlers
+{
+	public partial class HybridWebViewHandler : ViewHandler<IHybridWebView, AWebView>
+	{
+		private HybridWebViewJavaScriptInterface? _javaScriptInterface;
+
+		protected override AWebView CreatePlatformView()
+		{
+#if DEBUG
+			var logger = MauiContext!.Services!.GetService<ILogger<HybridWebViewHandler>>() ?? NullLogger<HybridWebViewHandler>.Instance;
+			logger.LogInformation("HybridWebViewHandler: CreatePlatformView Android WebView");
+#endif
+			var platformView = new HybridPlatformWebView(this, Context!)
+			{
+				LayoutParameters = new LayoutParams(LayoutParams.MatchParent, LayoutParams.MatchParent)
+			};
+
+			platformView.Settings.DomStorageEnabled = true;
+			platformView.Settings.SetSupportMultipleWindows(true);
+
+			// Note that this is a per-app setting and not per-control, so if you enable
+			// this, it is enabled for all Android WebViews in the app.
+			AWebView.SetWebContentsDebuggingEnabled(enabled: true); // TODO: Get from setting
+
+			platformView.Settings.JavaScriptEnabled = true;
+
+			_javaScriptInterface = new HybridWebViewJavaScriptInterface(this);
+			platformView.AddJavascriptInterface(_javaScriptInterface, "hybridWebViewHost");
+
+			return platformView;
+		}
+
+		private sealed class HybridWebViewJavaScriptInterface : Java.Lang.Object
+		{
+			private readonly HybridWebViewHandler _hybridWebViewHandler;
```

Comment:
Do you have info on how to test this?

Reply:
Was pointed to this test file: https://github.com/dotnet/maui/blob/main/src/Controls/tests/DeviceTests/Memory/MemoryTests.cs

And this PR: https://github.com/dotnet/maui/pull/23540/files

Reply:
This should be addressed now with the recent commits.

Reply:
@mattleibow I think you'll be happy to hear that since I added the test, it was able to locate several memory leaks due to lack of WeakRef's. I've updated the code to be extremely weakly referenced and the memory leak test passes on all platforms!